sycl/test-e2e/Graph/RecordReplay/record_inconsistent_contexts.cpp

```diff
@@ -0,0 +1,27 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out 2>&1 | FileCheck %s
+//
+// CHECK: begin_recording called for a queue whose context differs from the graph context
+
+// This test check that an expection is thrown when we try to
+// record a graph whose context differs from the queue context.
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  property_list Properties{property::queue::in_order()};
+  context InOrderContext(Properties);
+
+  exp_ext::command_graph Graph{InOrderContext, Queue.get_device()};
+
+  try {
+    Graph.begin_recording(Queue);
+  } catch (exception &Exception) {
+    std::cout << Exception.what() << std::endl;
+  }
```

Comment:
We should probably be catching the `sycl::exception` here and checking that the error code is what we expect.

Also applies to the other test.

Reply:
If you look at how Ben's checked for exceptions in his open PR, we don't use any `CHECK` directives, just compare against the error code `assert(e.code() == sycl::errc::invalid);`

Reply:
Ok. I updated these tests.